tests/test_metadata_parser.py

```diff
@@ -0,0 +1,40 @@
+from metadata_table import MetadataHeader, Var
+
+example_table = """
+[ccpp-table-properties]
+  name = <name>
+  type = scheme
+  relative_path = path
+  dependencies = a.f,b.f
+
+[ccpp-arg-table]
+  name = <name>
+  type = scheme
+[ im ]
+  standard_name = horizontal_loop_extent
+  long_name = horizontal loop extent, start at 1
+  units = index
+  type = integer
+  dimensions = ()
+  intent = in
+"""
+
+
+def test_MetadataHeader_parse_table(tmpdir):
+    path = str(tmpdir.join("table.meta"))
+    with open(path, "w") as f:
+        f.write(example_table)
+
+    table1, table2 = MetadataHeader.parse_metadata_file(path)
+
+    # check first table
+    assert table1.name == "<name>"
+    assert table1.type == "scheme"
+    assert table1.dependencies == ["path/a.f", "path/b.f"]
+
+    # check second table
+    assert table2.name == "<name>"
+    assert table2.type == "scheme"
+    (im_data,) = table2.variable_list()
```

Comment:
What would happen here if the variable list happened if `table2.variable_list()` returned zero or more than one value? Wouldn't it throw an exception?
To me, it would be more robust to search for the variable (e.g., using `find_variable`) and also check that the length of the variable list is one.

Reply:
> What would happen here if the variable list happened if table2.variable_list() returned zero or more than one value? Wouldn't it throw an exception?

This is the intended effect, yes. This is a way to assert the length is one and unpack the value at the same time. The example table only has one variable so my understanding was that parsing >=1 or 0 values would be a failure.  Is your concern that this error is not raised within an assertion? I am used to that, but am open to changing to
```
assert len(table2._variable_list()) == 1
im_data = table1.variable_list()[0]
```
if you think it's more clear.

Reply:
Thanks for the explanation. In general, I think custom error messages are better.
As for your change, this is more clear. However, since these tests will be replaced with the more extensive set coming from feature/capgen, I am fine merging this version.

Reply:
Okay. Thanks.